controllers/record-utils.js

```diff
@@ -51,12 +70,240 @@ var createRecordByJSON = function(data, callback) {
       }
     }
   }
+
   // filter out any unwanted fields
   data._meta = _.pick(data._meta, required.concat(optional));
-  // no need to pass the content type as nano.request defaults to json.
-  request({ body: data }, callback);
+
+  // remove fields that start with underscore except _meta
+  _.omit(data, function(v, k) {
+    return k !== '_meta' && String(k).startsWith('_');
+  });
+
+  // Using `_meta` property for non-form data.
+  if (data._meta) {
+    doc.reported_date = data._meta.reported_date;
+    doc.form = smsparser.getFormCode(data._meta.form);
+    doc.from = data._meta.from;
+    doc.locale = data._meta.locale || doc.locale;
+  }
+
+  def = config.getForm(doc.form);
+
+  if (!def) {
+    return callback(new Error('Form not found: ' + doc.form));
+  }
+
+  // For now only save string and number fields, ignore the others.
+  // Lowercase all property names.
+  _.each(data, function(v, k) {
+    if (['string', 'number'].indexOf(typeof data[k]) >= 0) {
+      form_data[k.toLowerCase()] = data[k];
+    }
+  });
+
+  create(getDataRecord(doc, form_data), callback);
+};
+
+/**
+ * @param {String} form - form code
+ * @param {Object} form_data - parsed form data
+ * @returns {String} - Reporting Unit ID value (case insensitive)
+ * @api private
+ */
+var getRefID = function(form, form_data) {
+  var def = config.getForm(form),
+      val;
+  if (!def || !def.facility_reference) {
+    return;
+  }
+  val = form_data && form_data[def.facility_reference];
+  if (val && typeof val.toUpperCase === 'function') {
+    return val.toUpperCase();
+  }
+  return val;
+};
+
+/**
+ * @param {Object} options from initial POST
+ * @param {Object} form_data - parsed form data
+ * @returns {Object} - data record
+ * @api private
+ */
+var getDataRecord = function(options, form_data) {
+
+  var form = options.form,
+      def = config.getForm(form);
+
+  var record = {
+   // _id: req.uuid,
+    type: 'data_record',
+    // todo maybe change libphonenumber to only take default_country_code
+    from: libphonenumber.normalize(config.settings, options.from) || options.from,
```

Comment:
Currently to update the shrinkwrap all you have to do is `npm install google-libphonenumber@<x.y.z>`. With transitive dependencies I'm pretty sure that won't work so you'll have to update and publish medic-libphonenumber and then `npm install medic-libphonenumber@<x.y.z>` so a few extra steps to the current release process. It's possible that if `medic-libphonenumber` depends on a range then when we can update the transitive dependency with a standard `npm install` but I want to confirm this before we rely on having our own wrapper module. I really don't want to "patch the shrinkwrap" - that way leads to madness, because npm will blow away the patch any time shrinkwrap is updated.